R/preprocess.R

```diff
@@ -492,6 +492,27 @@ enw_latest_data <- function(obs) {
   return(latest_data[])
 }
 
+#' Filter observations to restrict the maximum reporting delay
+#'
+#' @description `r lifecycle::badge("deprecated")`
+#' @return A `data.frame` filtered so that dates by report are less than or
+#' equal the reference date plus the maximum delay.
+#'
+#' @inheritParams enw_add_incidence
+#' @inheritParams enw_preprocess_data
+#' @importFrom lifecycle deprecate_stop
+#' @family preprocess
+#' @export
+enw_filter_delay <- function(obs, max_delay, timestep = "day") {
```

Comment:
I think we might want to stick with using `deprecate_warn` for now. Unless you feel strongly?